src/app/pages/movie-details-page/movie-details-page.component.ts

```diff
@@ -113,4 +146,62 @@ export class MovieDetailsPageComponent implements OnInit {
     }
   }
 
+  async generateViewModal(e: any) {
+    
+  }
+  ViewedStatus=false;
```

Comment:
Variables needs to be declared at the top of the class and they need to be a specific type